java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -415,9 +415,9 @@ public static MinionServer createSystemProfile(User creator, String systemName,
         }
 
         Set<String> hwAddrs = hwAddress.map(a -> singleton(a)).orElse(emptySet());
-        if (findMatchingEmptyProfile(hostname, hwAddrs).isPresent()) {
-            throw new IllegalStateException("System(s) with hostname '" + hostname + "' or HW address '" +
-                    hwAddress + "' already exists.");
+        List<MinionServer> matchingProfiles = findMatchingEmptyProfiles(hostname, hwAddrs);
+        if (!matchingProfiles.isEmpty()) {
+            throw new SystemsExistException(matchingProfiles.stream().map(p -> p.getId()).collect(Collectors.toList()));
```

Comment:
I'm missing `import com.redhat.rhn.manager.system.SystemsExistException;`

Shouldn't it be imported as well?

Reply:
No, it's in the same java package (also, if it was missing, the code wouldn't even compile).